lib/config.mk.in

```diff
@@ -5,6 +5,6 @@ CUDA_OBJS := @CUDA_OBJS@
 DEFINE_MACROS := @DEFINE_MACROS@
 prefix := @prefix@
 exec_prefix := @exec_prefix@
-CFLAGS := @MKCFLAGS@ -I@includedir@
+CFLAGS := @CPPFLAGS@ @MKCFLAGS@ -I@includedir@
```

Comment:
Unfortunately that does work because AC_CHECK_HEADER uses the CPPFLAGS variable by default.

```
$ CFLAGS="-I/opt/local/include" LDFLAGS="-L/opt/local/lib" ./configure
checking for clang... clang
checking for nvcc... nvcc
checking neon... no
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether clang accepts -g... yes
checking for clang option to accept ISO C89... none needed
checking for sgemm in -lblas... yes
checking how to run the C preprocessor... clang -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking png.h usability... yes
checking png.h presence... no
configure: WARNING: png.h: accepted by the compiler, rejected by the preprocessor!
configure: WARNING: png.h: proceeding with the compiler's result
checking for png.h... yes
checking jpeglib.h usability... yes
checking jpeglib.h presence... no
configure: WARNING: jpeglib.h: accepted by the compiler, rejected by the preprocessor!
configure: WARNING: jpeglib.h: proceeding with the compiler's result
checking for jpeglib.h... yes
checking fftw3.h usability... yes
checking fftw3.h presence... no
configure: WARNING: fftw3.h: accepted by the compiler, rejected by the preprocessor!
configure: WARNING: fftw3.h: proceeding with the compiler's result
checking for fftw3.h... yes
checking linear.h usability... no
checking linear.h presence... no
checking for linear.h... no
checking tesseract/capi.h usability... no
checking tesseract/capi.h presence... no
checking for tesseract/capi.h... no
checking Accelerate/Accelerate.h usability... yes
checking Accelerate/Accelerate.h presence... yes
checking for Accelerate/Accelerate.h... yes
checking libavcodec/avcodec.h usability... no
checking libavcodec/avcodec.h presence... no
checking for libavcodec/avcodec.h... no
checking libavformat/avformat.h usability... no
checking libavformat/avformat.h presence... no
checking for libavformat/avformat.h... no
checking libswscale/swscale.h usability... no
checking libswscale/swscale.h presence... no
checking for libswscale/swscale.h... no
checking dispatch/dispatch.h usability... yes
checking dispatch/dispatch.h presence... yes
checking for dispatch/dispatch.h... yes
checking for dispatch_apply in -ldispatch... no
checking xmmintrin.h usability... yes
checking xmmintrin.h presence... yes
checking for xmmintrin.h... yes
checking for cos in -lm... yes
checking for cblas_dgemm in -lgslcblas... no
checking for gsl_blas_dgemm in -lgsl... no
checking cuda... yes, x86_64
-e 
 USE: clang
 COMPILE FLAGS: -fblocks -msse2 -D HAVE_CBLAS -D HAVE_LIBPNG -D HAVE_LIBJPEG -D HAVE_FFTW3 -D HAVE_ACCELERATE_FRAMEWORK -D USE_DISPATCH -D HAVE_SSE2 -D HAVE_CUDA
 LINK FLAGS: -lm -lblas -lpng -ljpeg -lfftw3 -lfftw3f -lpthread -framework Accelerate -lcuda -lcudart -lcublas -L/usr/local/cuda/lib64

configure: creating ./config.status
config.status: creating config.mk
```

with CPPFLAGS:

```
$ CPPFLAGS="-I/opt/local/include" LDFLAGS="-L/opt/local/lib" ./configure
checking for clang... clang
checking for nvcc... nvcc
checking neon... no
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether clang accepts -g... yes
checking for clang option to accept ISO C89... none needed
checking for sgemm in -lblas... yes
checking how to run the C preprocessor... clang -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking png.h usability... yes
checking png.h presence... yes
checking for png.h... yes
checking jpeglib.h usability... yes
checking jpeglib.h presence... yes
checking for jpeglib.h... yes
checking fftw3.h usability... yes
checking fftw3.h presence... yes
checking for fftw3.h... yes
checking linear.h usability... no
checking linear.h presence... no
checking for linear.h... no
checking tesseract/capi.h usability... no
checking tesseract/capi.h presence... no
checking for tesseract/capi.h... no
checking Accelerate/Accelerate.h usability... yes
checking Accelerate/Accelerate.h presence... yes
checking for Accelerate/Accelerate.h... yes
checking libavcodec/avcodec.h usability... no
checking libavcodec/avcodec.h presence... no
checking for libavcodec/avcodec.h... no
checking libavformat/avformat.h usability... no
checking libavformat/avformat.h presence... no
checking for libavformat/avformat.h... no
checking libswscale/swscale.h usability... no
checking libswscale/swscale.h presence... no
checking for libswscale/swscale.h... no
checking dispatch/dispatch.h usability... yes
checking dispatch/dispatch.h presence... yes
checking for dispatch/dispatch.h... yes
checking for dispatch_apply in -ldispatch... no
checking xmmintrin.h usability... yes
checking xmmintrin.h presence... yes
checking for xmmintrin.h... yes
checking for cos in -lm... yes
checking for cblas_dgemm in -lgslcblas... no
checking for gsl_blas_dgemm in -lgsl... no
checking cuda... yes, x86_64
-e 
 USE: clang
 COMPILE FLAGS: -fblocks -msse2 -D HAVE_CBLAS -D HAVE_LIBPNG -D HAVE_LIBJPEG -D HAVE_FFTW3 -D HAVE_ACCELERATE_FRAMEWORK -D USE_DISPATCH -D HAVE_SSE2 -D HAVE_CUDA
 LINK FLAGS: -lm -lblas -lpng -ljpeg -lfftw3 -lfftw3f -lpthread -framework Accelerate -lcuda -lcudart -lcublas -L/usr/local/cuda/lib64

configure: creating ./config.status
config.status: creating config.mk
```